src/hooks/resources/device.ts

```diff
@@ -576,3 +590,65 @@ async function checkSupervisorReleaseUpgrades(
 		}
 	}
 }
+
+async function setSupervisorReleaseResource(
+	api: sbvrUtils.PinejsClient,
+	deviceId: number,
+	supervisorVersion: string,
+) {
+	const device = (await api.get({
+		resource: 'device',
+		id: deviceId,
+		options: {
+			$expand: {
+				is_of__device_type: {
+					$select: 'id',
+				},
+			},
+			// if the device already has a supervisor_version, just bail.
+			$filter: {
+				supervisor_version: { $eq: null },
+			},
+			$select: ['id', 'supervisor_version'],
+		},
+	})) as AnyObject;
+
+	if (device.length === 0) {
+		return null;
+	}
+
+	const supervisorRelease = (await api.get({
+		resource: 'supervisor_release',
+		options: {
+			$select: ['id'],
+			$filter: {
+				supervisor_version: `v${supervisorVersion}`,
+				is_for__device_type: {
+					$any: {
+						$alias: 'dt',
+						$expr: {
+							dt: {
+								id: device.is_of__device_type[0].id,
+							},
+						},
+					},
+				},
+			},
+		},
+	})) as AnyObject[];
+
+	if (supervisorRelease.length !== 0) {
+		await api.patch({
```

Comment:
I'd go with
```suggestion
	if (supervisorRelease.length === 0) {
		return;
	}
	await api.patch({
```
to match the short-circuiting style done previously and to stay consistent - it also makes it clear that if it's 0 then we do nothing and return whereas in the current version there could technically be code after the `if`